Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -349,6 +350,30 @@ void ValidateProducedLayout()
             }
         }
 
+        [Feature(Features.Symlink)]
+        [Feature(Features.OpaqueDirectory)]
+        [TheoryIfSupported(requiresSymlinkPermission: true)]
+        public void DirectorySymlinksInOutputDirectoryShouldSuccess()
+        {    
+            AbsolutePath rootDirAbsPath = CreateUniqueObjPath($"{SealDirectoryKind.SharedOpaque}.framework");
+            string rootDir = rootDirAbsPath.ToString(Context.PathTable);
+            System.Diagnostics.Debugger.Launch();
+
+            // schedule producer pip: produce opaque directory with a bunch of symlinks
+            var operations = GetLayoutProducingOperationsWithDummyReadFile(rootDir, "producer");
+            var producerPipBuilder = CreatePipBuilder(operations);
+            producerPipBuilder.ToolDescription = StringId.Create(Context.StringTable, "producer");
+
+            // if dirKind is an opaque output, then just declare the root directory as an opaque output;
+            // otherwise, add all outputs explicitly and schedule a seal directory pip to seal them.
+            Process producerPip;
+            DirectoryArtifact outputDirArtifact;
+            producerPipBuilder.AddOutputDirectory(rootDirAbsPath, kind: SealDirectoryKind.SharedOpaque);
+            producerPip = SchedulePipBuilder(producerPipBuilder).Process;
+            outputDirArtifact = producerPip.DirectoryOutputs.First();
+            ScheduleRunResult firstRunResult = RunScheduler().AssertSuccess();         

```

Comment:
unused variables: `firstRunResult`, `outputDirArtifact`.

In addition to expecting success, we should also specify what else should happen.  For example:
  - what happens when you clean the output folder and re-run the same pip?  Cache hit will happen but the directory symlink will not be restored on Windows.
  - also what happens when you re-run the pip in terms of scrubbing?  Is the directory symlink supposed to be scrubbed?  

All that said, I think it might be better to enable the existing `DirectorySymlinksInOutputDirectory` unit tests for Windows and then in that unit test have different assertions based on what operating system the test is running on

Reply:
Thanks again. Will try to reuse the test case.

Reply:
Tried to run DirectorySymlinksInOutputDirectory in windows. None of the condition has any luck to succeed. Checkback the test cases in DirectorySymlinkTests.cs, all test cases under this file are designed with requiresUnixBasedOperatingSystem=true. So I am wondering the history of the feature. Is it supposed to be different in windows and mscOs? I will spent a bit more time to understand the symlink feature and then back to the unit test here.

Reply:
It's not really that it is "supposed" to be different, it just currently is different because directory symlinks have not yet been fully implemented for Windows.  So my suggestion was to reuse the same test but add `if (OperatingSystemHelper.IsUnixOS)` around all those checks that don't pass on Windows so that we have one place where we can see what works and what doesn't work on Windows.  @narasamdya, what do you think?  Having a separate unit test is also fine with me.

Reply:
By running the existing test case for MAC, found out the behavior of the consuming directory symlink was not as expected. Without specified input of files in the output share opaque directory, BuildXL report read access violation. After offline check with @almiliMSFT, we agreed to limit the test only on the producer with testing the cache behavior.